test/ad_backends.jl

```diff
@@ -0,0 +1,34 @@
+@timed_testset "ad_backends" begin
+    DATA_DIR = joinpath("..", "data")
+    cheese = CSV.read(joinpath(DATA_DIR, "cheese.csv"), DataFrame)
+    f = @formula(y ~ (1 | cheese) + background)
+    m = turing_model(f, cheese)
+    # only running 2 samples to test if the different ADs runs
+    @timed_testset "ForwardDiff" begin
+        Turing.setadbackend(:forwarddiff)
+        chn = sample(seed!(123), m, NUTS(), MCMCThreads(), 10, 2)
+        @assert chn isa Chains
+    end
+    @timed_testset "Tracker" begin
+        using Tracker
+        Turing.setadbackend(:tracker)
+        chn = sample(seed!(123), m, NUTS(), MCMCThreads(), 10, 2)
+        @assert chn isa Chains
+    end
+    @timed_testset "Zygote" begin
+        using Zygote
+        Turing.setadbackend(:zygote)
+        chn = sample(seed!(123), m, NUTS(), MCMCThreads(), 10, 2)
+        @assert chn isa Chains
+    end
+    @timed_testset "ReverseDiff" begin
+        using ReverseDiff
+        Turing.setadbackend(:reversediff)
+        Turing.setrdcache(true)
+        chn = sample(seed!(123), m, NUTS(), MCMCThreads(), 10, 2)
+        @assert chn isa Chains
```

Comment:
@devmotion, is this enough to only test if the AD backends runs ok? 10 multithreaded samples?

Reply:
For a smoke test just sampling two or so samples should be sufficient. I don't think using multithreaded sampling provides any additional value. Setting a seed seems also not necessary if you don't care about the results apart from the type.

For ReverseDiff, `setrdcache(true)` should also not be necessary. In fact, one has to be careful when using it will not be correct for e.g. models with non-deterministic branches (that's a general limitation when you want to compile tapes with ReverseDiff - any function with branches, i.e., where in different runs the computations might take a different path, will be incorrect usually).

Reply:
But for future-proof sake we could leave the `setrdcache(true)` just to make sure it will sample with a RD tape?

Reply:
Any model that works with RD without tape and does not have any branches, should work with RD with tape correctly. There's no special things such as Memoization involved anymore.

Reply:
So I guess it does not matter :shrug: